File: frontend/Gorki/src/app/service/notification-service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../env/environment';
import { NotificationDTO } from '../model/ui/notification-dto';

@Injectable({ providedIn: 'root' })
export class NotificationService {
  private api = environment.apiHost;

  constructor(private http: HttpClient) {}

  getAllForUser(userId: number): Observable<NotificationDTO[]> {
    return this.http.get<NotificationDTO[]>(`${this.api}/notifications/user/${userId}`);
  }

  markAsRead(notificationId: number, userId: number): Observable<void> {
    return this.http.put<void>(`${this.api}/notifications/${notificationId}/read/${userId}`, {});
  }
}

File: mobile/app/src/main/java/ftn/mrs_team11_gorki/service/NotificationService.java
package ftn.mrs_team11_gorki.service;

import java.util.List;

import ftn.mrs_team11_gorki.dto.NotificationResponseDTO;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface NotificationService {

    @GET("api/notifications/user/{userId}")
    Call<List<NotificationResponseDTO>> getAllForUser(@Path("userId") long userId);

    @PUT("api/notifications/{id}/read/{userId}")
    Call<ResponseBody> markAsRead(@Path("id") long notificationId, @Path("userId") long userId);
}